src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -505,6 +505,18 @@ public void OpenComponent(int sequence, [DynamicallyAccessedMembers(Component)]
         OpenComponentUnchecked(sequence, componentType);
     }
 
+    /// <summary>
+    /// Appends a frame representing a component parameter.
+    /// </summary>
+    /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
+    /// <param name="name">The name of the attribute.</param>
+    /// <param name="value">The value of the attribute.</param>
+    public void AddComponentParameter(int sequence, string name, object? value)
```

Comment:
How does this approach work when the parameters are splatted?

Reply:
That calls `AddMultipleAttributes(int sequence, IEnumerable<KeyValuePair<string, object>>? attributes)` which doesn't have the same overload resolution issue.

If you're suggesting adding a `AddMultipleComponentParameters` method we could do but it would have no difference in behavior. I agree the aesthetics of still having to use `AddMultipleAttributes` for components are a bit strange, but if RenderTreeBuilder is only intended as a compiler target it's less of a priority.